### DISCUSSION TITLE
how to use async and await for utils function

### DISCUSSION THREAD

**Discussion by james-wu** [microsoft/RulesEngine#383] [Q&A]
is it possible to async and await for the utils function, something like

 ```
public static class Utils
    {
        public static async Task<bool> ValidateFromDbAsync(string country)
        {
             var result = await queryFromDbAsync(country);

            return result != null;
           
        }
    } 
```

`"Expression": "await Utils.ValidateFromDbAsync(input1.country) == true"
`

Please let me know if someone has any idea on this.

**Comment by abbasc52** [comment]
You can use async await in utils functions but not in expressions.
In Expression you would need to use `.Result` 
e.g. `"Expression": "Utils.ValidateFromDbAsync(input1.country).Result == true"`
